### PR TITLE
push db changes

### PR DESCRIPTION
**Pull request by ericshermancs**

### CODE REVIEW

File: src/core/grade_notifier.py [hunk 1]
```diff
@@ -46,23 +48,25 @@
...
 # Create .env file path.
 dotenv_path = join(constants.abs_repo_path(), '.env')
 
 # Load file from the path.
-load_dotenv(dotenv_path)
+load_dotenv()
 
 # Accessing variables.
 account_sid = os.getenv('TWILIO_SID')
 auth_token = os.getenv('TWILIO_AUTH_TOKEN')
```
Comment (Huddie) [inline src/core/grade_notifier.py]: space

File: src/core/grade_notifier.py [hunk 2]
```diff
@@ -225,23 +191,68 @@ def refresh():
...
+    while True:
+
+        cursor.execute('''SELECT * FROM Users WHERE lastUpdated < NOW() - INTERVAL 30 MINUTE ORDER BY lastUpdated DESC LIMIT 1''') # get top row from 
+        # cursor.execute('''SELECT * FROM Users LIMIT 1''')
+        try: 
+            row = cursor.next()
+            column_names = cursor.column_names
+
+            query_dict = {k: v for k,v in zip(column_names, row)}
+            pprint(query_dict)
```
Comment (Huddie) [inline src/core/grade_notifier.py]: remove

File: src/core/grade_notifier.py [hunk 3]
```diff
@@ -225,23 +191,68 @@ def refresh():
...
+
+            key = os.getenv('DB_ENCRYPTION_KEY').encode('utf-8')
+
+            f = Fernet(key)
+            password = f.decrypt(encrypted_password.encode('utf-8')).decode()
+
+            api = CUNYFirstAPI(query_dict['username'], password, query_dict['school'].upper())
+            api.login()
+
+            grade_result= refresh(api)
```
Comment (Huddie) [inline src/core/grade_notifier.py]: space

File: src/core/dbsetup.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import os
+import mysql.connector
+from dotenv import load_dotenv
+
+load_dotenv()
+
+ACCOUNT_PASSWORD = os.getenv('LOCALHOST_PASSWORD')
```
Comment (Huddie) [inline src/core/dbsetup.py]: LOCALHOST_PASSWORD should be DB_PASSWORD.. same for username

File: src/core/grade_notifier.py [hunk 5]
```diff
@@ -225,23 +191,68 @@ def refresh():
...
-            else None
-        if changelog is not None:
-            message = create_text_message(changelog)
-            send_text(message, user.get_number())
-            old_result = result
-        counter += 1
-        time.sleep(5 * 60)  # 5 min intervals
 
+    myconnector = mysql.connector.Connect(user=DB_USERNAME,
+        host='localhost',passwd=DB_PASSWORD)
```
[the remark this answers is not in the document]
Comment (Huddie) [reply src/core/grade_notifier.py]: ^^

File: src/core/grade_notifier.py [hunk 6]
```diff
@@ -225,62 +195,78 @@ def refresh():
...
-    with open(file_path) as oldfile:
-        for line in oldfile:
-            if stored_username not in line:
-                file += line
-    with open(file_path, 'w+') as newfile:
-        redacted_list = [username]
-        newfile = RedactedFile(newfile, redacted_list)
-        newfile.writelines(file)
+
+    myconnector = mysql.connector.Connect(user=DB_USERNAME,
```
Comment (Huddie) [inline src/core/grade_notifier.py]: extract 'localhost' as global variable

File: src/core/grade_notifier.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Huddie) [reply src/core/grade_notifier.py]: Whitespace

File: src/core/dbsetup.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+ACCOUNT_USERNAME = os.getenv('LOCALHOST_USERNAME')
+mydb = mysql.connector.connect(user=ACCOUNT_USERNAME,host='localhost', passwd=ACCOUNT_PASSWORD)
+mycursor = mydb.cursor()
+mycursor.execute('CREATE DATABASE IF NOT EXISTS GradeNotifier')
+mycursor.execute('USE GradeNotifier')
+mycursor.execute('CREATE TABLE IF NOT EXISTS Users (id INT AUTO_INCREMENT PRIMARY KEY, username VARCHAR(255), password VARCHAR(255), school VARCHAR(10), gradeHash VARCHAR(255), phoneNumber VARCHAR(255), lastUpdated DATETIME NOT NULL DEFAULT "1970-01-01 00:00:00", dateCreated DATETIME NOT NULL DEFAULT NOW())')
```
[the remark this answers is not in the document]
Comment (ericshermancs) [reply src/core/dbsetup.py]: keep for now, can remove later if we want

File: src/core/grade_notifier.py (hunk 6 above)
Comment (ericshermancs) [reply src/core/grade_notifier.py]: ok

File: src/core/dbsetup.py (hunk 4 above)
Comment (ericshermancs) [reply src/core/dbsetup.py]: same for host

**Pull request merged**